.teamcity/_self/projects/WebApp.kt

```diff
@@ -775,121 +775,17 @@ fun playwrightPrBuildType( targetDevice: String, buildUuid: String ): E2EBuildTy
-		param("env.CALYPSO_BASE_URL", "https://wpcalypso.wordpress.com")
-	},
-	buildFeatures = {
-		notifications {
-			notifierSettings = slackNotifier {
-				connection = "PROJECT_EXT_11"
-				sendTo = "#e2eflowtesting-notif"
-				messageFormat = verboseMessageFormat {
-					addStatusText = true
-				}
-			}
-			branchFilter = "+:<default>"
-			buildFailedToStart = true
-			buildFailed = true
-			buildFinishedSuccessfully = false
-			buildProbablyHanging = true
-		}
-	}
-)
-
-object AuthenticationE2ETests : E2EBuildType(
-	buildId = "calypso_WebApp_Calypso_E2E_Authentication",
-	buildUuid = "f5036e29-f400-49ea-b5c5-4aba9307c5e8",
-	buildName = "Authentication E2E Tests",
-	buildDescription = "Runs a suite of Authentication E2E tests.",
-	concurrentBuilds = 1,
-	testGroup = "authentication",
-	buildParams = {
-		param("env.VIEWPORT_NAME", "desktop")
-	},
-	buildFeatures = {
-		notifications {
-			notifierSettings = slackNotifier {
-				connection = "PROJECT_EXT_11"
-				sendTo = "#e2eflowtesting-notif"
-				messageFormat = verboseMessageFormat {
-					addStatusText = true
-				}
-			}
-			branchFilter = "+:<default>"
-			buildFailedToStart = true
-			buildFailed = true
-			buildFinishedSuccessfully = false
-			buildProbablyHanging = true
-		}
-	},
-	buildTriggers = {
-		schedule {
-			schedulingPolicy = cron {
-				hours = "*/6"
-			}
-			branchFilter = "+:<default>"
-			triggerBuild = always()
-			withPendingChangesOnly = true
-		}
-	}
-)
-
-object HelpCentreE2ETests : E2EBuildType(
-	buildId = "calypso_WebApp_Calypso_E2E_Help_Centre",
-	buildUuid = "a0e62582-8598-483e-8b82-9de540288f6d",
-	buildName = "Help Centre E2E Tests",
-	buildDescription = "Runs a suite of Help Centre E2E tests.",
-	testGroup = "help-centre",
-	buildParams = {
-		param("env.VIEWPORT_NAME", "desktop")
-	},
-	buildFeatures = {
-		notifications {
-			notifierSettings = slackNotifier {
-				connection = "PROJECT_EXT_11"
-				sendTo = "#e2eflowtesting-notif"
-				messageFormat = verboseMessageFormat {
-					addStatusText = true
-				}
-			}
-			branchFilter = "+:<default>"
-			buildFailedToStart = true
-			buildFailed = true
-			buildFinishedSuccessfully = false
-			buildProbablyHanging = true
-		}
-	},
-	buildTriggers = {
-		schedule {
-			schedulingPolicy = cron {
-				hours = "*/3"
-			}
-			branchFilter = "+:<default>"
-			triggerBuild = always()
-			withPendingChangesOnly = false
-		}
-	}
-)
-
-object KPIDashboardTests : BuildType({
-	id("calypso_WebApp_Calypso_E2E_KPI_Dashboard")
-	uuid = "441efac5-721a-4557-9448-9234e89fb6b1"
-	name = "Test build for KPI Dashboard project"
-	description = "Test build configuration for KPI dashboard."
+object PreReleaseE2ETests : BuildType({
+	id("calypso_WebApp_Calypso_E2E_Pre_Release")
+	uuid = "9c2f634f-6582-4245-bb77-fb97d9f16533"
+	name = "Pre-Release E2E Tests"
+	description = "Runs a pre-release suite of E2E tests against trunk on staging, intended to be run after PR merge, but before deployment to production."
```

Comment:
Explanation: in order to get Allure connected and the dashboard data generation happening sooner rather than later, I made the decision to temporarily revert the definition for the Pre-Release Tests to the older `BuildType` definition instead of using the `E2EBuildType` class.

This is because the `E2EBuildType` class only defines hard-coded steps: _prepare, run tests, collect results_. It isn't set up to accept an arbitrary step, or even to redefine what happens in the steps themselves.

I have a follow-up task https://github.com/Automattic/wp-calypso/issues/69801 for a big refactor of the `E2EBuildType` class so that it can be extended as necessary.

Reply:
I added a comment over there about how we do it with the wpcom plugins

Reply:
Thanks, will take a look. I'll likely spin that into its own separate task, in context of a (possible) bigger TeamCity configuration rethink.